### PR TITLE
Rename flatten_folders to season_folders and fix #4170

### LINKED ISSUE #4170: When adding a new show into the Medusa database even if the option is checked to use Season folders after adding the show it becomes unchecked.
### Before submitting your issue:

Enable debug logging in Medusa settings, reproduce the error (be sure to disable after the bug is fixed)

**Branch/Commit:**master
**OS:**unRAID 6.5.1 using linuxserver/medusa
**What you did:** Added a new or existing show into Medusa's database
**What happened:**When adding a new show to the database even if the group season's folder option is checked after you add the show it becomes unchecked. Resulting in the show not using Season folders. Started a week or so ago just found out today.
**What you expected:**
**Logs:**
```
PASTE LOGS HERE
```

### PR DESCRIPTION
**Pull request by medariox**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: medusa/server/api/v1/core.py [hunk 1]
```diff
@@ -1965,8 +1965,8 @@ def __init__(self, args, kwargs):
         self.initial, args = self.check_params(args, kwargs, 'initial', None, False, 'list', list(quality_map))
         self.archive, args = self.check_params(args, kwargs, 'archive', None, False, 'list',
                                                list(quality_map).remove('unknown'))
-        self.flatten_folders, args = self.check_params(args, kwargs, 'flatten_folders',
-                                                       bool(app.FLATTEN_FOLDERS_DEFAULT), False, 'bool', [])
+        self.season_folders, args = self.check_params(args, kwargs, 'season_folders',
+                                                       bool(not app.FLATTEN_FOLDERS_DEFAULT), True, 'bool', [])
```
Comment (sharkykh) [inline medusa/server/api/v1/core.py]: I suggest [bumping the API version up](https://github.com/pymedusa/Medusa/blob/feature/rename-flatten-folders/medusa/server/api/v1/core.py#L101) due to these changes.

File: medusa/server/api/v1/core.py [hunk 2]
```diff
@@ -1797,8 +1797,8 @@ def run(self):
                 raise ApiError('Status Prohibited')
             app.STATUS_DEFAULT = self.status
 
-        if self.flatten_folders is not None:
-            app.FLATTEN_FOLDERS_DEFAULT = int(self.flatten_folders)
+        if self.season_folders is not None:
+            app.FLATTEN_FOLDERS_DEFAULT = int(not self.season_folders)
```
Comment (sharkykh) [inline medusa/server/api/v1/core.py]: As we discussed on Slack, the `app` variable name should change too:
`FLATTEN_FOLDERS_DEFAULT => SEASON_FOLDERS_DEFAULT`

File: medusa/server/web/home/add_shows.py [hunk 3]
```diff
@@ -297,7 +297,7 @@ def existingShows(self):
     def addShowByID(self, indexername=None, seriesid=None, show_name=None, which_series=None,
                     indexer_lang=None, root_dir=None, default_status=None,
                     quality_preset=None, any_qualities=None, best_qualities=None,
-                    flatten_folders=None, subtitles=None, full_show_path=None,
+                    season_folders=None, subtitles=None, full_show_path=None,
                     other_shows=None, skip_show=None, provided_indexer=None,
                     anime=None, scene=None, blacklist=None, whitelist=None,
                     default_status_after=None, default_flatten_folders=None,
```
Comment (sharkykh) [inline medusa/server/web/home/add_shows.py]: Remove `default_flatten_folders` - it isn't used.
Should also change these lines:
https://github.com/pymedusa/Medusa/blob/2a35e2405272bab897b349c23f0787450d65f5c0/themes-default/slim/static/js/add-shows/init.js#L140-L141
to: (remove the comment line)
```js
season_folders: $('#season_folders').prop('checked'), // eslint-disable-line camelcase
```

This should actually fix #4170...

File: medusa/tv/series.py [hunk 4]
```diff
@@ -1957,7 +1957,7 @@ def save_to_db(self):
                           'quality': self.quality,
                           'airs': self.airs,
                           'status': self.status,
-                          'flatten_folders': self.flatten_folders,
+                          'flatten_folders': self.season_folders,
```
Comment (sharkykh) [inline medusa/tv/series.py]: You're saving to the database, value should be inverted.
```py
'flatten_folders': not self.season_folders,
```

Comment (medariox) [reply medusa/tv/series.py]: Great catch, fixed.

File: medusa/server/api/v1/core.py (hunk 1 above)
Comment (sharkykh) [reply medusa/server/api/v1/core.py]: Also,

File: medusa/server/api/v1/core.py [hunk 5]
```diff
@@ -2051,8 +2051,8 @@ def __init__(self, args, kwargs):
         self.initial, args = self.check_params(args, kwargs, 'initial', None, False, 'list', list(quality_map))
         self.archive, args = self.check_params(args, kwargs, 'archive', None, False, 'list',
                                                list(quality_map).remove('unknown'))
-        self.flatten_folders, args = self.check_params(args, kwargs, 'flatten_folders',
-                                                       bool(app.FLATTEN_FOLDERS_DEFAULT), False, 'bool', [])
+        self.season_folders, args = self.check_params(args, kwargs, 'season_folders',
+                                                       bool(not app.FLATTEN_FOLDERS_DEFAULT), True, 'bool', [])
```
Comment (sharkykh) [inline medusa/server/api/v1/core.py]: `flake8` check failed:
```
/home/travis/build/pymedusa/Medusa/medusa/server/api/v1/core.py:2055:56: E127 continuation line over-indented for visual indent
```

File: tests/legacy/test_lib.py [hunk 6]
```diff
@@ -74,7 +74,7 @@ def create_test_cache_folder():
 
 app.showList = []
 app.QUALITY_DEFAULT = 4  # hdtv
-app.FLATTEN_FOLDERS_DEFAULT = 0
+app.SEASON_FOLDERS_DEFAULT = 0
```
Comment (sharkykh) [inline tests/legacy/test_lib.py]: Change this to `1` for consistency.

**Pull request merged**